streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -603,7 +603,11 @@ public void checkpoint() {
         try {
             checkpointFile.write(checkpointingOffsets);
         } catch (final IOException e) {
-            log.warn("Failed to write offset checkpoint file to [{}]", checkpointFile, e);
+            log.warn("Failed to write offset checkpoint file to [{}]." +
+                " This may occur if OS cleaned the state.dir in case when it located in /tmp directory." +
+                " You can change location for state.dir to resolve problem." +
+                " This can also occur due to running multiple instances on the same machine using the same state dir.",
```

Comment:
Hmmm... Not sure if I can follow? Why?

Also, if you run multiple instances on the same machine, it seems to be the same as running multiple threads in the same `KafkaStreams` client?

Also, if you configure different state directories, local state cannot be shared what might be wasteful?

Can you elaborate on this sentence?

Reply:
@mjsax you can't run multiple instances on the same machine with the same state.dir. For one thing, the locking mechanism is per-process. If you run two different instances then you can get an active task on one instance and the corresponding standby on another. They would each think they owned the lock for that task directory, and concurrently access it (leading to the FileNotFoundException if one of them deletes the checkpoint, for example)

Reply:
Currently, running multiple threads within an instance is _not_ the same as running multiple instances on the same machine in the same state dir. I suppose technically if you could configure the two instances to have the same client UUID then the task assignor would make sure not to assign the same active/standby task to anyone on that machine, but I think there might be some unexpected side effects to running two separates instances with the same UUID.

Anyways, if you're aware of the problem enough to configure the client Ids directly, you should be aware enough to just configure to use separate state.dirs. I don't think I buy the argument about it being wasteful just because there's really no reason to ever do this in a production app. But people seem to do it in testing all the time, and run into this issue, hence the warning here

Reply:
Thanks for clarification @ableegoldman!